### ISSUE TITLE
Unable to use scripts created in Code Editor

### ISSUE THREAD

**Issue by LeXXik** [playcanvas/editor#1129]
1. Open Code Editor
2. Create a script
3. Try to attach it to entity in Editor

Expected: the script is shown in the list of available scripts
Actual: the script is missing

No errors in console.

https://github.com/playcanvas/editor/assets/5677782/07bbc51d-650b-4a23-acc3-0c4d06f6ccfa

**Comment by LeXXik** [comment]
Probably related: #1079

**Comment by mvaligursky** [comment]
Thanks, urgently investigating.

**Comment by LeXXik** [comment]
Some notes:

- The script is not appearing even after a browser refresh.
- It appears after you parse it in Editor, though.

**Comment by marklundin** [comment]
Thanks for flagging this @LeXXik, we'll be pushing a fix  for this in the next release

**Comment by marklundin** [comment]
Fixed and deployed

**Comment by ideathingdev** [comment]
This is happening again on 1.74.0.
Parsing doesn't help

**Comment by willeastcott** [comment]
Yes, if you create the script in the Code Editor, it does not auto-parse. But if you create the script in the main Editor's Assets panel, it _does_ auto-parse. @marklundin, please take a look.

**Comment by Maksims** [comment]
Also, due to delay of scripts parsing (it takes time from the moment of creating script asset till it is parsed), please add progress (like assets with pipeline jobs) onto the asset to indicate it is awaiting parsing finish.
And select an asset immediately, not with dealy. As it is very annoying when selecting an entity to add script, then creating a script and 2-4 seconds after Editor changes selection, while you already in the middle of something else.

**Comment by marklundin** [comment]
This was working, but looks like the `parseScriptCallback` [was removed](https://github.com/playcanvas/editor-ui/pull/379/files#diff-539e100f32f811bea6515c278f81a4e9f083543b61384d74a8de9ab1b9fa3e17L69) in the [API updates](https://github.com/playcanvas/editor-ui/pull/379) and so no new scripts are parsed in the editor. @kpal81xd was this supposed to be set elsewhere?

**Comment by marklundin** [comment]
Closing. This has been fixed
